google/cloud/bigquery/CMakeLists.txt

```diff
@@ -0,0 +1,176 @@
+# ~~~
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ~~~
+
+set(DOXYGEN_PROJECT_NAME "Google Cloud BigQuery C++ Client")
+set(DOXYGEN_PROJECT_BRIEF "A C++ Client Library for Google Cloud BigQuery")
+set(DOXYGEN_PROJECT_NUMBER "${PROJECT_VERSION} (Beta)")
+set(DOXYGEN_EXAMPLE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/samples
+                         ${CMAKE_CURRENT_SOURCE_DIR}/quickstart)
+set(DOXYGEN_EXCLUDE_PATTERNS "*/*_test.cc")
+set(DOXYGEN_EXCLUDE_SYMBOLS "internal" "bigquery_internal" "bigquery_testing"
+                            "examples")
+set(DOXYGEN_PREDEFINED "GOOGLE_CLOUD_CPP_NS=v1")
+
+include(GoogleCloudCppCommon)
+
+# TODO(#4146) - remove FPHSA_NAME_MISMATCHED manipulation on next absl release
+set(FPHSA_NAME_MISMATCHED Threads) # Quiet warning caused by Abseil
+find_package(absl CONFIG REQUIRED)
+unset(FPHSA_NAME_MISMATCHED)
+
+# configure_file(version_info.h.in ${CMAKE_CURRENT_SOURCE_DIR}/version_info.h)
+add_library(
+    google_cloud_cpp_bigquery # cmake-format: sort
+    big_query_read_client.gcpcxx.pb.cc
+    big_query_read_client.gcpcxx.pb.h
```

Comment:
These file paths seem overly verbose. It seems like this one should just be called "read_client". The "big_query" part is redundant given the path.

We may also want to reconsider the need and value of the `.gcpxx.pb.` part in the include paths. It's not clear whether this naming is good/bad for our customers.